src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -208,7 +209,7 @@ void BaseCouplingScheme::receiveData(const m2n::PtrM2N &m2n, const DataMap &rece
 void BaseCouplingScheme::receiveDataForWindowEnd(const m2n::PtrM2N &m2n, const DataMap &receiveData)
 {
   const double oldComputedTimeWindowPart = _computedTimeWindowPart;
-  _computedTimeWindowPart += getTimeWindowSize(); // such that getTime() in receiveData returns time at end of window
+  _computedTimeWindowPart += _nextTimeWindowSize; // such that getTime() in receiveData returns time at end of window
   this->receiveData(m2n, receiveData);            // receive data for end of window
   _computedTimeWindowPart = oldComputedTimeWindowPart;
```

Comment:
Could we make this more explicit to improve readability?

Reply:
Took care about this in [d3542d4](https://github.com/precice/precice/pull/1887/commits/d3542d4e7596c2bec954f86cf93d9ca00673eef7).